test/reputation-system/reputation-mining-client/client-auto-functionality.js

```diff
@@ -427,12 +463,22 @@ process.env.SOLIDITY_COVERAGE
             }, 30000);
           });
 
-          // Forward time
-          await forwardTime(600, this);
-          // Good client should now realise it can timeout bad submission
+          disputeRound = await repCycle.getDisputeRound(0);
+          badEntry = disputeRound[badIndex];
+          goodEntry = disputeRound[goodIndex];
 
-          await goodClientInvalidateOpponent;
+          const noChallengeCompleted = noEventSeen(repCycleEthers, "ChallengeCompleted");
+          const noHashInvalidated = noEventSeen(repCycleEthers, "HashInvalidated");
+
+          await noChallengeCompleted;
```

Comment:
Remind me why we don't just `await` these promises directly?

Reply:
Because it's testing the auto functionality. If we awaited directly where they're created, the actions to trigger them have not yet taken place on chain so the test would time out. If we moved the creation to where they're awaited, then the tests become spotty, as the thing the promises are waiting for might have happened before they've been set up.